### DISCUSSION TITLE
How to put articles in a subdirectory

### DISCUSSION THREAD

**Discussion by grifferz** [welpo/tabi#331] [Q&A] [answered]
Hi, I am completely new to Zola and tabi and I'm afraid I am completely confused even after trying to read all documentation of both.

I'd like to put my articles in the directory `content/posts/` so as per tabi's documentation:

> If you want to serve your blog posts from a different path, such as `blog/`, add a `section_path` in the `[extra]` section of `content/_index.md`

So, my `content/_index.md` looks like:

```
+++
title = "Latest posts"
sort_by = "date"
template = "index.html"

[extra]
header.title = "Site title"
header.img = "img/header.png"
header.img_alt = "header image"
section_path = "posts/"
+++
```

and at `content/posts/foo/index.md` I have:

```
+++
title = "foo"
date = "2024-06-18"
+++

Test.
```

At this point `zola serve` gives me:

```
Error: Failed to render section '/home/andy/src/test-site/content/_index.md'
Error: Reason: Failed to render 'index.html' (error happened in a parent template)
Error: Reason: Function call 'get_section' failed
Error: Reason: Section `posts/` not found.
```

Okay, so I thought maybe that's because there is no `content/posts/_index.md` to make that a section. Having no real idea what to put into such a file, I tried with this, which I also gleaned from somewhere in the tabi documentation:

```
+++
paginate_by = 5
path = "/posts"
title = "Posts"
sort_by = "date"
template = "section.html"
insert_anchor_links = "left"

[extra]
show_previous_next_article_links = true
+++
```

The error remains the same, however. What am I misunderstanding? A pointer to documentation about this would be much appreciated.

Assuming I can get that working, I will actually want to put articles in further sub-directories by year, e.g. `content/posts/2024/2024-06-18-bar/index.md`. Reading around I think I will need to make the section `content/posts/2024` (and every other year-subdirectory) "transparent" so that things paginate properly. Will that work?

I understand that neither the year nor the rest of the date date will appear in the eventual URL (that will in this example come out as `/posts/bar`) but that is fine; I only want this for organisational purposes in my file system. Do I understand this correctly?

Thanks!

**Comment by welpo** [comment, marked as answer]
Hi there!

Thanks for not giving up on tabi & Zola, for reading the docs, and for the detailed message!

It's correct, you did need `content/posts/_index.md`.

The site is failing to build because of this line in `content/_index.md`:

```
section_path = "posts/"
```

It's missing the path to the `_index.md` file. Try changing it to:

```md
section_path = "posts/_index.md"
```

That should fix the error; everything else looks good! 

> A pointer to documentation about this would be much appreciated.

I've [updated README.md with a comment](https://github.com/welpo/tabi/commit/7d386283ff808576a7149a96223be3564984d783) explicitly stating it needs to include the full path to the index file.

---

I've never tried using the date subdirectory format, but I know others have. `transparent` —which passes the pages of that section to its parent section— will come in handy for this, yeah.

Give it a go and don't hesistate to reach out for any issues you encounter!

**Comment by grifferz** [comment]
Thank you very much! I was able to get it to work and am making good progress with this now.
